third_party/amd/backend/compiler.py

```diff
@@ -261,7 +259,8 @@ def make_ttgir(mod, metadata, options):
         passes.ttgpuir.add_optimize_dot_operands(pm, True)
         passes.ttgpuir.add_remove_layout_conversions(pm)
         passes.ttgpuir.add_reduce_data_duplication(pm)
-        if os.environ.get("TRITON_HIP_USE_IN_THREAD_TRANSPOSE", "0") == "1":
+        is_attention_variant = options.instruction_sched_variant.lower() == "attention"
+        if os.environ.get("TRITON_HIP_USE_IN_THREAD_TRANSPOSE", "0") == "1" or is_attention_variant:
```

Comment:
Don't bundle this together for now--we want to enable `TRITON_HIP_USE_IN_THREAD_TRANSPOSE` on by default soon separately.